### PR TITLE
add version check

### PR DESCRIPTION
**Pull request by alex-Arc**

### CODE REVIEW

File: src/v3/connection.ts [hunk 1]
```diff
@@ -205,6 +210,16 @@ export function connect(self: OnTimeInstance, ontime: OntimeV3): void {
...
+				case 'version': {
+					clearTimeout(versionTimeout as NodeJS.Timeout)
+					if (semver.satisfies(semver.coerce(payload) ?? '', '>=3.0.0')) {
+						self.updateStatus(InstanceStatus.Ok, payload)
+					} else {
+						self.updateStatus(InstanceStatus.ConnectionFailure, `Incompatible version ${payload}`)
+						ws?.close()
+					}
+					break
+				}
```
Comment (cpvalente) [inline src/v3/connection.ts:222]: Although I see how this is nice for users, I personally dont feel that is worth adding an external dependency

We could do a simple check for no check at all

Also, does the upgrade run every time a user instantiates the module?

Comment (alex-Arc) [reply src/v3/connection.ts:222]: upgrade runs on every restart i think

this logic is for when the connection starts and is not part of the upgrade stuff

so maybe just split on `.` look on `[0]` and check for it to be 3

start WS (set connect timeout) -> on connect -> get version (set version timeout)-> on version responds -> check version

Comment (cpvalente) [reply src/v3/connection.ts:222]: Hi @josephdadams , I am not entirely familiar what is the best approach for updates here
We have a new major release in Ontime (v3) which has breaking changes towards the APIs

We have also decided that we would not maintain support for multiple Ontime versions in the companion module as we did previously.

Would you mind giving us some feedback here?

Comment (josephdadams) [reply src/v3/connection.ts:222]: Are you anticipating most of your users will upgrade?

Comment (josephdadams) [reply src/v3/connection.ts:222]: Then just log something to the log “You can obtain an upgrade at…” etc. and then when you update the status: “Unsupported version: see log.”

**Review by cpvalente — APPROVED** [review]

**Comment by alex-Arc** [comment]
companion release 3.3
should we get #51 merged and the I can bump the version in this PR to v4.0.0 ?

**Pull request merged**